protob/messages-eos.proto

```diff
@@ -0,0 +1,265 @@
+    * Structure representing auth account
+    */
+    message EosAuthorizationAccount {
+        optional EosPermissionLevel account = 1;
+        optional uint32 weight = 2;
+    }
+
+    /**
+    * Structure representing auth delays
+    */
+    message EosAuthorizationWait {
+        optional uint32 wait_sec = 1;
+        optional uint32 weight = 2;
+    }
+
+    /**
+    * Structure representing authorization settings
+    */
+    message EosAuthorization {
+        optional uint32 threshold = 1;
+        repeated EosAuthorizationKey keys = 2;
+        repeated EosAuthorizationAccount accounts = 3;
+        repeated EosAuthorizationWait waits = 4;
+    }
+
+     /**
+    * Structure representing the common part of every action
+    */
+    message EosActionCommon {
+        optional uint64 account = 1;          // Contract name
+        optional uint64 name = 2;             // Action name
+        repeated EosPermissionLevel authorization = 3;
+    }
+     /**
+    * Structure representing transfer data structure
+    */
+    message EosActionTransfer {
+        optional uint64 sender = 1;             // Asset sender
+        optional uint64 receiver = 2;
+        optional EosAsset quantity = 3;                     
+        optional string memo = 4;
+    }
+     /**
+    * Structure representing delegation data structure
+    */
+    message EosActionDelegate {
+        optional uint64 sender = 1;
+        optional uint64 receiver = 2;
+        optional EosAsset net_quantity = 3;    // Asset format '1.0000 EOS'
+        optional EosAsset cpu_quantity = 4;    // Asset format '1.0000 EOS'
+        optional bool transfer = 5;            // Transfer delegated tokens or not.
+    }
+     /**
+    * Structure representing the removal of delegated resources from `sender`
+    */
+    message EosActionUndelegate {
+        optional uint64 sender = 1;
+        optional uint64 receiver = 2;
+        optional EosAsset net_quantity = 3;   // Asset format '1.0000 EOS'
+        optional EosAsset cpu_quantity = 4;   // Asset format '1.0000 EOS'
+    }
+    /**
+    * Structure representing fallbalck if undelegate wasnt executed automaticaly.
+    */
+    message EosActionRefund {
+        optional uint64 owner = 1;
+    }
+
+     /**
+    * Structure representing buying RAM operation for EOS tokens
+    */
+    message EosActionBuyRam {        
+        optional uint64 payer = 1;
+        optional uint64 receiver = 2;
+        optional EosAsset quantity = 3;   // Asset format '1.0000 EOS'
+    }
+     /** 
+    * Structure representing buying bytes according to RAM market price.
+    */
+    message EosActionBuyRamBytes {
+        optional uint64 payer = 1;
+        optional uint64 receiver = 2;
+        optional uint32 bytes = 3;       // Number of bytes
+    }
+     /** 
+    * Structure representing sell RAM
+    */
+    message EosActionSellRam {
+        optional uint64 account = 1;
+        optional sint64 bytes = 2;       // Number of bytes
```

Comment:
Should bytes be a `uint64` here?

https://github.com/EOSIO/eos/blob/905e7c85714aee4286fa180ce946f15ceb4ce73c/contracts/eosio.system/eosio.system.abi#L150